js/base-util.js

```diff
@@ -49,33 +50,27 @@ function baseUtil () {
 		});
 	};
 
+	d20plus.ut.injectCode = (object, method, injectedCode) => {
```

Comment:
not a fan of this, as it effectively changes the return value of the original method

The `libWrapper` Foundry module has a good system we should steal: the injected function is passed `(originalFunction, ...originalParams)` (rather than `params` as you have here) and can therefore decide whether to call `originalFunction(...originalParams)` (effectively passing-through) or not as it sees fit

Reply:
I've rewritten it a bit. Didn't find the Foundry method you were referring to, but I hope I got it right

Reply:
not quite, here's an example of what I was talking about:

```js
(() => {
	const injectCode = (object, method, injectedCode) => {
		const originalFunction = object[method].bind(object);

		object[method] = (...initParams) => {
			return injectedCode.bind(object)(originalFunction, ...initParams);
		}
	};

	const ThingToPatch = {
		_greeting: "hello",

		theirMethod (text) {
			console.log(this._greeting, text)
		}
	}

	let isFoo = false;

	function myInjection (boundMethod, ...params) {
		if (!isFoo) return boundMethod(...params)

		return console.error(this._greeting, ...params);
	}

	ThingToPatch.theirMethod("Alice")

	injectCode(ThingToPatch, "theirMethod", myInjection);

	ThingToPatch.theirMethod("Bob")
	isFoo = true
	ThingToPatch.theirMethod("Clarence")
})()
```

the original method gets passed in to the injected method; the injected method can then choose whether to run or not run (or e.g. run, and then modify the output of,) the original method

Reply:
I've kept both patterns, with a simple bool parameter to use the more primitive one.
Ability to simply write e.g. `injectCode(obj, method, console.log, true)` is too tempting to switch completely to full injection.